src/gui/rss/automatedrssdownloader.cpp

```diff
@@ -47,10 +50,35 @@
 #include "ui_automatedrssdownloader.h"
 #include "automatedrssdownloader.h"
 
+class AutomatedRssDownloader::DownloadRuleListMatchState
+{
+public:
+    DownloadRuleListMatchState();
+    ~DownloadRuleListMatchState();
+    void clear();
+
+public:
+    QSet<QPair<QString, QString >> m_treeListEntries;
+};
+
+AutomatedRssDownloader::DownloadRuleListMatchState::DownloadRuleListMatchState()
+{
+}
+
+AutomatedRssDownloader::DownloadRuleListMatchState::~DownloadRuleListMatchState()
+{
+}
+
+void AutomatedRssDownloader::DownloadRuleListMatchState::clear()
+{
+    m_treeListEntries.clear();
+}
+
 AutomatedRssDownloader::AutomatedRssDownloader(const QWeakPointer<Rss::Manager> &manager, QWidget *parent)
     : QDialog(parent),
     ui(new Ui::AutomatedRssDownloader),
-    m_manager(manager), m_editedRule(0)
+    m_manager(manager), m_editedRule(0),
+    m_ruleMatcher(new AutomatedRssDownloader::DownloadRuleListMatchState())
```

Comment:
Take note that the lines that you didn't chagne in this diff are also wrong in the initialization list.
(comma goes in the start to reduce noise in the diff when adding new fields)